librz/syscall/syscall.c

```diff
@@ -287,27 +277,22 @@ RZ_API RzList *rz_syscall_list(RzSyscall *s) {
 	return list;
 }
 
-/* io and sysregs */
-RZ_API const char *rz_syscall_get_io(RzSyscall *s, int ioport) {
+/**
+ * \brief Searches the sysregs SDB database inside librz/reg/d for the type (mmio/reg) and
+ * 		  returns its value as a string
+ *
+ * \param s reference to RzSyscall
+ * \param type reference to the type of key: mmio/reg
+ * \param port reference to the io port
+ */
+RZ_API const char *rz_sysreg_get(RzSyscall *s, const char *type, ut64 port) {
```

Comment:
Overall looks good, but I suggest to load SDB once, during the binary loading of the matching architecture, e.g. in the hashtable and query it instead during the analysis. This way, in case of many queries it will work much faster, especially for complex analysis loops.